src/iinit.cpp

```diff
@@ -43,27 +44,30 @@ const char *AUTH_OPENID_SCHEME = "openid";
  */
 int
 mkrodsdir() {
-    char dirName[NAME_LEN];
-    int mode;
-    char *getVar;
-#ifdef windows_platform
-    getVar = iRODSNt_gethome();
-#else
-    getVar = getenv( "HOME" );
-#endif
-    rstrcpy( dirName, getVar, NAME_LEN );
-    rstrcat( dirName, "/.irods", NAME_LEN );
-    mode = 0700;
-#ifdef _WIN32
-    iRODSNt_mkdir( dirName, mode );
-#else
-    int error_code = mkdir( dirName, mode );
-    int errsv = errno;
-    if ( error_code != 0 && errsv != EEXIST ) {
-        rodsLog( LOG_NOTICE, "mkdir failed in mkrodsdir with error code %d", error_code );
+    namespace fs = std::filesystem;
+
+    fs::path irods_dir{std::getenv("HOME")};
+    irods_dir / ".irods";
+
+    std::error_code dir_ec;
+    fs::create_directory(irods_dir, dir_ec);
+    if (dir_ec) {
+      std::cerr
+          << "create_directory failed in mkrodsdir with the following error: ["
+          << dir_ec.message() << "]\n";
+      return -1;
     }
```

Comment:
What happens if the directory already exists?

Reply:
Double-checked [cppreference for `create_directory(...)`](https://en.cppreference.com/w/cpp/filesystem/create_directory) and it should not report an error if the directory exists.

> 1) ... If the function fails because p resolves to an existing directory, no error is reported. Otherwise on failure an error is reported.